lib/units/storage/amazons3.js

```diff
@@ -0,0 +1,197 @@
+var http = require('http')
+var util = require('util')
+var path = require('path')
+
+var express = require('express')
+var validator = require('express-validator')
+var bodyParser = require('body-parser')
+var formidable = require('formidable')
+var Promise = require('bluebird')
+var uuid = require('node-uuid')
+var fs = require('fs')
+
+var lifecycle = require('../../util/lifecycle')
+var logger = require('../../util/logger')
+var requtil = require('../../util/requtil')
+
+var AWS = require('aws-sdk')
+
+module.exports = function(options) {
+  var log = logger.createLogger('storage:s3')
+  , app = express()
+  , server = http.createServer(app)
+  , credentials = new AWS.SharedIniFileCredentials({
+      profile: options.profile
+    })
+  , bucket = options.Bucket
+  , s3
+
+  // TODO: remove later
+  log.debug(options)
+
+  AWS.config.credentials = credentials;
+  s3 = new AWS.S3(options)
+
+  app.set('strict routing', true)
+  app.set('case sensitive routing', true)
+  app.set('trust proxy', true)
+
+  app.use(bodyParser.json())
+  app.use(validator())
+
+  function putObject(plugin, uriorpath, name) {
+    return new Promise(function(resolve, reject) {
+      var id = uuid.v4()
+      var key = id
+      var filename = name? name: path.basename(uriorpath)
+      var rs = fs.createReadStream(uriorpath)
```

Comment:
I removed whole ```pp.post('/s/download/:plugin', ...```. but, I want to remain ```getHref``` for readable. It is also used on ```app.post('/s/upload/:plugin',...``` to make response. is it ok?